### PR TITLE
CTCP-446 Add Contacts Telephone Number View

### PR DESCRIPTION
**Pull request by carrielies**

### CODE REVIEW

File: app/controllers/traderDetails/holderOfTransit/ContactTelephoneNumberController.scala [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  actions: Actions,
+  val controllerComponents: MessagesControllerComponents,
+  view: ContactTelephoneNumberView
+)(implicit ec: ExecutionContext)
+    extends FrontendBaseController
+    with I18nSupport {
+
+  def onPageLoad(lrn: LocalReferenceNumber, mode: Mode): Action[AnyContent] = actions.requireData(lrn) {
+    implicit request =>
+      request.userAnswers.get(ContactNamePage) match {
```
Comment (jackheslop96) [inline app/controllers/traderDetails/holderOfTransit/ContactTelephoneNumberController.scala:48]: This is where the getMandatoryPage action builder is useful. See address controller for an example. Applies to onSubmit as well

File: app/pages/traderDetails/holderOfTransit/AddContactPage.scala [hunk 2]
```diff
@@ -16,15 +16,24 @@
...
   override def path: JsPath = HolderOfTransitSection.path \ toString
 
   override def toString: String = "addContact"
 
-  //TODO - add clean up for changing answer to false
+  override def cleanup(value: Option[Boolean], userAnswers: UserAnswers): Try[UserAnswers] =
+    value match {
+      case Some(false) =>
+        userAnswers.remove(ContactNamePage)
+        userAnswers.remove(ContactTelephoneNumberPage)
```
Comment (jackheslop96) [inline app/pages/traderDetails/holderOfTransit/AddContactPage.scala:36]: `userAnswers.remove(ContactNamePage).flatMap(_.remove(ContactTelephoneNumberPage))`

File: conf/messages.en [hunk 3]
```diff
@@ -206,3 +206,12 @@ traderDetails.holderOfTransit.tirIdentification.checkYourAnswersLabel = TIR hold
 traderDetails.holderOfTransit.tirIdentification.hint = AAA/999/99999.
 traderDetails.holderOfTransit.tirIdentification.error.required = Enter the TIR holder’s identification number
 traderDetails.holderOfTransit.tirIdentification.error.invalidFormat = The TIR holder’s identification number must start with 3 letters, a forward slash, 3 numbers and another forward slash followed by up to 9 numbers
+
+traderDetails.holderOfTransit.contactTelephoneNumber.title = What is {0}’s telephone number?
+traderDetails.holderOfTransit.contactTelephoneNumber.heading = What is {0}’s telephone number?
+traderDetails.holderOfTransit.contactTelephoneNumber.checkYourAnswersLabel = {0}’s telephone number?
```
Comment (jackheslop96) [inline conf/messages.en:212]: I suggest this should be `Contact’s telephone number`

File: app/controllers/traderDetails/holderOfTransit/ContactTelephoneNumberController.scala (hunk 1 above)
Comment (carrielies) [reply app/controllers/traderDetails/holderOfTransit/ContactTelephoneNumberController.scala]: Done

**Review by jackheslop96 — APPROVED** [review]

**Pull request merged**